### ISSUE TITLE
only limited set of types processed

### ISSUE THREAD

**Issue by rtmie** [prometheus/collectd_exporter#31]
When I query the metrics endpoint of this it does not include many of the collectd data I am collecting.
On the nodes I am running collect the following plugins are enabled:
 
   `[root@compute1 collectd.d]# ls 
    cpu.conf  disk.conf  interface.conf  libvirt.conf  load.conf  memory.conf  network.conf  swap.conf      syslog.conf` 

However when I query the end point on my prometheus server I only see the following (e.g. no virt stats). I temporarily set up an influxdb with collectd listener and it gets all of the stats, so they are being sent. Is there some additional configuration that I am missing?

```
[robm@sysmgr ~]$ curl http://localhost:9103/metrics|grep -v '^#'`

 Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  7184  100  7184    0     0   928k      0 --:--:-- --:--:-- --:--:-- 1002k`

collectd_exporter_build_info{branch="cut-0.3.0",goversion="go1.6.2",revision="3abb95c",version="0.3.1"} 1
collectd_last_push_timestamp_seconds 0
go_gc_duration_seconds{quantile="0"} 7.452400000000001e-05
go_gc_duration_seconds{quantile="0.25"} 0.000101897
go_gc_duration_seconds{quantile="0.5"} 0.000103992
go_gc_duration_seconds{quantile="0.75"} 0.000108567
go_gc_duration_seconds{quantile="1"} 0.000231744
go_gc_duration_seconds_sum 0.011762737
go_gc_duration_seconds_count 110
go_goroutines 13
go_memstats_alloc_bytes 2.322928e+06
go_memstats_alloc_bytes_total 3.4490968e+08
go_memstats_buck_hash_sys_bytes 1.449911e+06
go_memstats_frees_total 222582
go_memstats_gc_sys_bytes 210944
go_memstats_heap_alloc_bytes 2.322928e+06
go_memstats_heap_idle_bytes 3.334144e+06
go_memstats_heap_inuse_bytes 2.6624e+06
go_memstats_heap_objects 6883
go_memstats_heap_released_bytes_total 0
go_memstats_heap_sys_bytes 5.996544e+06
go_memstats_last_gc_time_seconds 1.4685134677145672e+09
go_memstats_lookups_total 2321
go_memstats_mallocs_total 229465
go_memstats_mcache_inuse_bytes 1200
go_memstats_mcache_sys_bytes 16384
go_memstats_mspan_inuse_bytes 14280
go_memstats_mspan_sys_bytes 32768
go_memstats_next_gc_bytes 4.194304e+06
go_memstats_other_sys_bytes 524609
go_memstats_stack_inuse_bytes 294912
go_memstats_stack_sys_bytes 294912
go_memstats_sys_bytes 8.526072e+06
http_request_duration_microseconds{handler="prometheus",quantile="0.5"} 2603.717
http_request_duration_microseconds{handler="prometheus",quantile="0.9"} 3856.373
http_request_duration_microseconds{handler="prometheus",quantile="0.99"} 3956.057
http_request_duration_microseconds_sum{handler="prometheus"} 711660.7530000001
http_request_duration_microseconds_count{handler="prometheus"} 230
http_request_size_bytes{handler="prometheus",quantile="0.5"} 305
http_request_size_bytes{handler="prometheus",quantile="0.9"} 305
http_request_size_bytes{handler="prometheus",quantile="0.99"} 305
http_request_size_bytes_sum{handler="prometheus"} 67972
http_request_size_bytes_count{handler="prometheus"} 230
http_requests_total{code="200",handler="prometheus",method="get"} 230
http_response_size_bytes{handler="prometheus",quantile="0.5"} 1594
http_response_size_bytes{handler="prometheus",quantile="0.9"} 1601
http_response_size_bytes{handler="prometheus",quantile="0.99"} 1601
http_response_size_bytes_sum{handler="prometheus"} 416246
http_response_size_bytes_count{handler="prometheus"} 230
process_cpu_seconds_total 0.94
process_max_fds 1.048576e+06
process_open_fds 8
process_resident_memory_bytes 9.621504e+06
process_start_time_seconds 1.46850995405e+09
process_virtual_memory_bytes 1.654784e+07
```

**Comment by brian-brazil** [comment]
Those are all internal stats of the collectd exporter. Are you sure you're sending data to it?

**Comment by rtmie** [comment]
I thought that they looked internal.
Following your tip I looked again and the issue is the way I ran the docker contianer:

`sudo docker run -d -p 9103:9103 -p 25826:25826 prom/collectd-exporter -collectd.listen-address=":25826"`

Because protocol is not specified in my docker call for port 25826 it will defaul to being a TCP rather than UDP port.

Changing the docker run to
`
sudo docker run -d -p 9103:9103 -p 25826:25826/udp prom/collectd-exporter -collectd.listen-address=":25826"`

will cause the container to map UDP on the collectd port and now I am seeing the additional collectd_* stats.

It might be worth updating the readme.md with specific call for the docker container. If I get time tomorrow  I will do a PR for that.
Thanks.
